server/src/main/java/au/org/aodn/ogcapi/server/common/RestExtApi.java

```diff
@@ -27,10 +27,16 @@ public class RestExtApi {
     protected RestExtService restExtService;
     @Autowired
     protected Search searchService;
+
     @GetMapping(path="/autocomplete")
-    public ResponseEntity<List<String>> getAutocompleteSuggestions(@RequestParam String input) throws java.lang.Exception {
-        return searchService.getAutocompleteSuggestions(input);
+    public ResponseEntity<List<String>> getAutocompleteSuggestions(
+            @RequestParam String input,
+            //categories is an optional parameter, if not provided, the method will return suggestions from all categories
+            @RequestParam(required = false) List<String> categories
```

Comment:
if you prefer to keep the autocomplete then the requestparam should be cql filter string instead of List<String> categories?

Reply:
I'm not sure if the use of CQL query can be relevant for the implementation of "search_as_you_type" since they are not the same thing.

e.g "/collections?q=input='imos'&category='temperature'"

The return type also be different. 

I have very detailed comments in the method to construct the "search_as_you_type" query.